vignettes/tetrapod-biodiversity.Rmd

```diff
@@ -0,0 +1,384 @@
+
+- *Albertosaurus sarcophagus*
+- *Ankylosaurus* sp.
+- *Ceratopsidae* indet.
+- *Ornithominus* sp.
+- *Tyrannosaurus rex* 
+
+A filter for species-level identifications would reduce the species richness to just two (*A. sarcophagus* and *T. rex*). However, none of these clades are nested within one another, so each of the indeterminately identified occurrences represents at least one species not already represented in the dataset. `tax_unique` is designed to deal with such taxonomic data and would retain all seven 'species' in this example. For further details see `?tax_unique`.
+
+This function can be especially useful for when looking at alpha diversity (local richness). Let's apply it to our dataset by filtering it to genus-level. 
+
+```{r}
+# filter to genus-level:
+tet_genera <- tax_unique(occdf = maj_tetrapods, genus = "genus", family = "family",
+                         order = "order", class = "class", resolution = "genus")
+```
+
+We can compare this to the genera from the original dataset to see how many genera were added by using `tax_unique()`:
+
+```{r}
+# How many unique genera before:
+length(unique(maj_tetrapods$genus))
+
+# How many unique genera after:
+length(unique(tet_genera$unique_name))
+
+```
+
+As you can see we've added three occurrences that otherwise would have been ignored! To see more specifically how the function has performed, take a look at row 806, which records an indeterminate occurrence of the family *Melosauridae*. This occurrence now has a unique name of 'Melosauridae indet.'
+
+```{r}
+tet_genera[806, ]
+
+```
+
+Now, let's use this function to extract the unique genera for each collection (=locality) with `group_apply`: 
+
+```{r}
+coll_genera <- group_apply(occdf = maj_tetrapods,
+                           group = c("collection_no"),
+                           fun = tax_unique,
+                           genus = "genus",
+                           family = "family",
+                           order = "order",
+                           class = "class",
+                           resolution = "genus")
+
+# see structure of the new dataframe:
+str(coll_genera)
+```
+
+Next, let's count the number of unique genera per collection:
+
+```{r}
+coll_taxa <- aggregate(coll_genera$unique_name, by=list(coll_genera$collection_no), FUN=length)
+
+# rename column names:
+colnames(coll_taxa) <- c("collection_no", "n_taxa")
+
+# take a look at the new dataframe
+head(coll_taxa)
+```
+
+To plot an alpha diversity (local richness) plot, we will need the ages of intervals in millions of years. Since the numeric ages may have changed since we downloaded the data from the PBDB, we'll use `look_up` to get the numeric ages for the occurrences within `tetrapods` dataset based on their assigned interval names:
+
+```{r}
+# get new numeric ages for named intervals
+# the default is to use the Geological Time Scale 2020
+occdf <- look_up(maj_tetrapods)
```

Comment:
I think this should probably be introduced before binning data earlier on? Note that not all intervals can be matched using just GTS2020. You might want to also include the built-in interval key?